groups/script.js

```diff
@@ -76,46 +90,85 @@ const memberAddRoleBody = {
-groupsData?.forEach((item) => {
-  const group = document.createElement('li');
-  group.setAttribute('id', item.roleid);
-  group.classList.add('group-role');
-  const formattedRoleName = removeGroupKeywordFromDiscordRoleName(
-    item.rolename,
-  );
 
-  //If searchValue present, filter out the list
-  if (searchValue) {
-    group.style.display = formattedRoleName
-      .toUpperCase()
-      .includes(searchValue.toUpperCase())
-      ? ''
-      : 'none';
-  }
+const renderGroups = () => {
+  groupRoles.innerHTML = null;
+  groupsData?.forEach((item) => {
+    const group = document.createElement('li');
+    group.setAttribute('id', item.roleid);
+    group.classList.add('group-role');
+    const formattedRoleName = removeGroupKeywordFromDiscordRoleName(
+      item.rolename,
+    );
+
+    //If searchValue present, filter out the list
+    if (searchValue) {
+      group.style.display = formattedRoleName
+        .toUpperCase()
+        .includes(searchValue.toUpperCase())
+        ? ''
+        : 'none';
+    }
 
-  const groupname = document.createElement('p');
-  groupname.classList.add('group-name');
-  groupname.setAttribute('id', `name-${item.roleid}`);
+    const groupname = document.createElement('p');
+    groupname.classList.add('group-name');
+    groupname.setAttribute('id', `name-${item.roleid}`);
 
-  if (item.memberCount !== null && item.memberCount !== undefined) {
-    groupname.setAttribute('data-member-count', item.memberCount);
-  }
+    if (item.memberCount !== null && item.memberCount !== undefined) {
+      groupname.setAttribute('data-member-count', item.memberCount);
+    }
 
-  groupname.textContent = formattedRoleName;
+    groupname.textContent = formattedRoleName;
 
-  const createdBy = createAuthorDetailsDOM(
-    item.firstName,
-    item.lastName,
-    item.image,
-  );
+    const createdBy = createAuthorDetailsDOM(
+      item.firstName,
+      item.lastName,
+      item.image,
+    );
 
-  group.appendChild(groupname);
-  group.appendChild(createdBy);
-  groupRoles.appendChild(group);
-});
+    group.appendChild(groupname);
+    group.appendChild(createdBy);
+    groupRoles.appendChild(group);
+  });
+};
+
+const giveABForCompariosn = (a, b, field) => {
+  switch (field) {
+    case 'date._seconds':
+      if (a.date && a.date._seconds) {
+        return [a.date._seconds, b.date._seconds];
+      }
+    case 'memberCount':
+      return [a.memberCount, b.memberCount];
+    default:
+      return [0, 0];
+  }
+};
+
+function onDropdownClick(ev) {
+  const clickedOptionsId = ev.target.dataset.list;
+  const fieldToSortBy = SortByFields.find(
+    (field) => field.id === clickedOptionsId,
+  );
+  groupsData.sort((firstObj, secondObj) => {
+    const [a, b] = giveABForCompariosn(
+      firstObj,
+      secondObj,
+      fieldToSortBy.fieldName,
+    );
+    if (a > b) {
+      return -1;
+    } else if (b < a) {
+      return 1;
+    }
+    return 0;
+  });
```

Comment:
Would something like this work here?
```groupsData.sort((firstObj, secondObj) => firstObj[fieldToSortBy.fieldName] - secondObj[fieldToSortBy.fieldName])```

The code can be more concise this way.

Reply:
Okay, this doesn't work because of the field might not be present case.